include/rocksdb/compaction_filter.h

```diff
@@ -59,6 +60,10 @@ class CompactionFilter {
   };
 
   virtual ~CompactionFilter() {}
+  static const char* Type() { return "CompactionFilter"; }
+  static Status CreateFromString(const ConfigOptions& config_options,
+                                 const std::string& name,
+                                 const CompactionFilter** result);
```

Comment:
Can we use std::shared_ptr<CompactionFilter>* result here?

Reply:
A CompactionFilter (unlike a CompactionFilterFactory) is stored as a raw pointer in the Options infrastructure.  This signature matches that (singleton) usage pattern.